circuits/cpp/barretenberg/py/src/estimation/main.py

```diff
@@ -0,0 +1,71 @@
+from flavors import GoblinUltra
+from circuit import CircuitKZG
+from goblin import Goblin
+from protogalaxy import FoldingVerifier, Decider
+from utils import DataLog
+
+# App circuits an kernel circuits are in a "stack" and all
+# are assumed to have a fixed size
+LOG_N_IN_STACK = 13
+
+# A Goblin instance conists of three circuit instances, one for each proof output
+# by the client. A circuit instance contains the data of _estimates of_:
+#   - the size of the circuit 
+#   - the maximum amount of memory used during construction of the circuit 
+#   - the time to construct a recursive verifier circuit's MSMs using UltraCircuitConstructor
```

Comment:
e.g. with the ECCVM PR, we have a special ECCVMCircuitConstructor that builds a circuit to evaluate the MSMs required by the recursive verifier (and/or the protogalaxy folding verifier)

Reply:
Yeah, but the ECCVM circuit in the estimation tool will have an estimate of the amount of time it would take an UltraHonk circuit to verify an ECCVM proof. That's the model here: the base layer consists of an UltraHonk circuit that verifies two Goblin proofs (==6 different individual proofs).